openedx/features/enterprise_support/api.py

```diff
@@ -664,3 +669,35 @@ def insert_enterprise_pipeline_elements(pipeline):
     insert_point = pipeline.index('social_core.pipeline.social_auth.load_extra_data')
     for index, element in enumerate(additional_elements):
         pipeline.insert(insert_point + index, element)
+
+
+@enterprise_is_enabled()
+def unlink_enterprise_user_from_idp(request, user, idp_backend_name):
+    """
+    Un-links learner from their enterprise identity provider
+    Args:
+        request (wsgi request): request object
+        user (User): user who initiated disconnect request
+        idp_backend_name (str): Name of identity provider's backend
+
+    Returns: None
+
+    """
+    enterprise_customer = enterprise_customer_for_request(request)
+    if user and enterprise_customer:
+        enabled_providers = Registry.get_enabled_by_backend_name(idp_backend_name)
+        provider_ids = [enabled_provider.provider_id for enabled_provider in enabled_providers]
```

Comment:
As per our discussion this can yield lot of provider ids and due to that `EnterpriseCustomerIdentityProvider` query below may become expensive. 

I think something like below can solve the purpose, We also need to ensure that identity provider returned by `EnterpriseCustomerIdentityProvider` should have a `tpa-saml` backend.
```python
enterprise_customer_idps = EnterpriseCustomerIdentityProvider.objects.filter(
    enterprise_customer__uuid=enterprise_customer['uuid']
).
```

Reply:
@muhammad-ammar I don't think select query is going to be expensive since we are filtering by customer `enterprise_customer__uuid=enterprise_customer['uuid']` first(which is indexed by default due to onetoone relationship) and then filtering the providers. I did not release that while we discussing it.

Reply:
In that case we can merge the PR